### PR TITLE
Bug fix dist model save

### PR DESCRIPTION
**Pull request by wangz10**
*Issue #, if available:* #535

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/trainer/gsgnn_trainer.py [hunk 1]
```diff
@@ -192,11 +192,12 @@ def save_model(self, model, epoch, i, save_model_path):
         '''
         barrier()
         if save_model_path is not None:
-            assert isinstance(model.module, (GSgnnModel, GSgnnModelBase)), \
```
Comment (zheng-da) [inline python/graphstorm/trainer/gsgnn_trainer.py:195]: there are integration tests for the standalone mode. Can you trigger model saving in integration test of standalone mode?

Comment (zheng-da) [reply python/graphstorm/trainer/gsgnn_trainer.py:195]: here is the integration test. https://github.com/awslabs/graphstorm/blob/main/tests/end2end-tests/graphstorm-nc/test.sh#L30

Comment (wangz10) [reply python/graphstorm/trainer/gsgnn_trainer.py:195]: done

File: python/graphstorm/eval/evaluator.py [hunk 2]
```diff
@@ -326,6 +326,11 @@ def best_iter_num(self):
         """
         return self._best_iter
 
+    @property
+    def history(self):
+        """ Evaluation history"""
+        return self._history
```
Comment (zheng-da) [inline python/graphstorm/eval/evaluator.py:332]: maybe revert this change and add this to the tutorial?

**Review by zheng-da — APPROVED** [review]
